src/sysrepo.h

```diff
@@ -972,12 +972,16 @@ int sr_oper_delete_item_str(sr_session_ctx_t *session, const char *path, const c
 
 /**
  * @brief Prepare to discard nodes matching the specified xpath in the operational datastore before applying
- * the other push oper data of this sessions. Usable only for ::SR_DS_OPERATIONAL datastore. These changes are applied
+ * the other push oper data of this session. Usable only for ::SR_DS_OPERATIONAL datastore. These changes are applied
  * only after calling ::sr_apply_changes().
  *
  * Creates an opaque node `discard-items` in the `sysrepo` YANG module namespace with @p xpath used as the value.
  * Such a node can be a part of the edit in ::sr_edit_batch() and will discard nodes like this function does.
  *
+ * This affects only operational data of lower order. For correct behaviour, `sr_set_oper_changes_order()` must be called
+ * from all sessions that push operational data to the module to explicitly set the order.
+ * `sr_discard_items` will not give consistent results with automatically generated order.
+ *
```

Comment:
I disagree with most of this, it should be formulated as a recommendation, not a necessity. You will get "correct" behavior even without explicitly setting the order.

Reply:
Done